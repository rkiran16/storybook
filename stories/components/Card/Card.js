import { html } from "lit-html";
import { IconButtonWithDropdown } from "../IconButtonWithDropdown/IconButtonWithDropdown";
import { Link } from "../Link/Link";

const Card = () => {
  return html`
    <div class="card">
      <div class="card-header d-flex justify-content-between">
          <div>
            <h5 class="card-title">Lorem Ipsum Dolor</h5>
            <p class="mb-0">Lorem ipsum dolor sit amet, consectetur adipiscing elit</p>
          </div>
          ${IconButtonWithDropdown({textDirection: 'horizontal', inverse: false, disabled: false, iconName: 'more_vert', buttonText: 'Actions', className: 'p-0'})}
      </div>
      <div class="card-body">
        <p class="card-text pb-4">
           
        </p>
      </div>
      <div class="card-footer">
       ${Link({})}
      </div>
    </div>
  `;
}

export const CardFull = ({size}) => {
  const sizeClass = `col-md-12 col-lg-${size}`;
  return html`
    <div class="container">
        <div class="row">
          <div class=${sizeClass}>
              ${Card()}
          </div>
        </div>
    </div>
  `;
}