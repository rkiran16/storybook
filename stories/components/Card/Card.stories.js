import { CardFull } from "./Card";

export default {
   title: "Components/Card",
   parameters: {
      layout: 'centered',
      zeplinLink: "https://app.zeplin.io/project/62902a463701ffb0e532df2b/screen/6307938941ea6181917cde3f",
   },
   argTypes: {
      size: {
         name: "Card Size",
         control: { type: 'select'},
          description: "Define the size of the Card",
          defaultValue: 6,
          options: [ 4, 6, 9, 12],
      }
   }
}

const Template = (args) => CardFull(args);

export const Standard = Template.bind({});