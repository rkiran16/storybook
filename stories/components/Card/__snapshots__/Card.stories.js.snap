// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Components/Card Standard 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "
    <div class=\\"container\\">
        <div class=\\"row\\">
          <div class=",
    ">
              ",
    "
          </div>
        </div>
    </div>
  ",
  ],
  "values": Array [
    "col-md-12 col-lg-6",
    Object {
      "_$litType$": 1,
      "strings": Array [
        "
    <div class=\\"card\\">
      <div class=\\"card-header d-flex justify-content-between\\">
          <div>
            <h5 class=\\"card-title\\">Lorem Ipsum Dolor</h5>
            <p class=\\"mb-0\\">Lorem ipsum dolor sit amet, consectetur adipiscing elit</p>
          </div>
          ",
        "
      </div>
      <div class=\\"card-body\\">
        <p class=\\"card-text pb-4\\">
           
        </p>
      </div>
      <div class=\\"card-footer\\">
       ",
        "
      </div>
    </div>
  ",
      ],
      "values": Array [
        Object {
          "_$litType$": 1,
          "strings": Array [
            "
    <div class=",
            " style=",
            ">
      <div class=\\"dropdown\\">
        <button
          type=\\"button\\"
          ?disabled=",
            "
          class=",
            "
          id=\\"dropdownIconButton\\" 
          data-bs-toggle=\\"dropdown\\" 
          aria-expanded=\\"false\\"
        >
          <span class=\\"material-icons ",
            "\\">",
            "</span>
          ",
            "
        </button>
        <ul class=\\"dropdown-menu dropdown-menu-end shadow-sm\\" aria-labelledby=\\"dropdownIconButton\\">
          <li tabIndex=\\"-1\\" class=\\"dropdown-item\\">Action </li>
          <li tabIndex=\\"-1\\" class=\\"dropdown-item\\"><a tabIndex=\\"-1\\" href=\\"#\\">Link</a></li>
          <li tabIndex=\\"-1\\" class=\\"dropdown-item\\">Another Action</li>
        </ul>
      </div>
    </div>
  ",
          ],
          "values": Array [
            "d-inline-block p-0",
            "",
            false,
            "dropdown-toggle btn btn-icon btn-icon-w-text horizontal flex-row-reverse ",
            "ms-1",
            "more_vert",
            "Actions",
          ],
        },
        Object {
          "_$litType$": 1,
          "strings": Array [
            "<a href=",
            " target=",
            ">",
            "</a>",
          ],
          "values": Array [
            "http://www.cardinalhealth.com",
            "_self",
            "Link action",
          ],
        },
      ],
    },
  ],
}
`;
